### PR TITLE
WooCommerce: Connect store address fields to state, persist to server

### PR DESCRIPTION
**Pull request by allendav**
See #14775 

To test, go to
- http://calypso.localhost:3000/store/{site}
- Make sure the name, address, address2, city, state, postcode and country fields are displayed.
- Note that they are editable. Make some changes.
- Make sure the Next button works
- After you click the Next button, verify you are taken to the "egg" view (i.e. the list of tasks and checkboxes)
- Note that after you've done "Next" that the set_store_address_during_initial_setup flag will be set on WPCOM for that site.
- To get back to the address view, you can “reset” the site using the developer console at  https://developer.wordpress.com/docs/api/console/ -- you’ll need to POST to /sites/siteID/calypso-preferences/woocommerce setting set_store_address_during_initial_setup to 0
- Make sure the changes you made are there
- Lastly, run npm run test-client client/extensions/woocommerce/state and make sure everything passes

Caveats:
- I'm not happy with the userBeganEditing flag in PreSetupView but without it, you can start editing and then lose your edits on a later prop wash.  Open to advice on how to resolve that.
- Site Title is not editable yet

### CODE REVIEW

File: client/extensions/woocommerce/app/dashboard/pre-setup-view.js [hunk 1]
```diff
@@ -10,46 +10,100 @@ import React, { Component, PropTypes } from 'react';
  * Internal dependencies
  */
 import AddressView from 'woocommerce/components/address-view';
+import {
+	areSettingsGeneralLoading,
+	getStoreLocation,
+} from 'woocommerce/state/sites/settings/general/selectors';
+import { errorNotice } from 'state/notices/actions';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/dashboard/pre-setup-view.js]: Flip this with `fetchSettingsGeneral` for alphabetical order.

File: client/extensions/woocommerce/app/dashboard/pre-setup-view.js [hunk 2]
```diff
@@ -75,16 +130,31 @@ class PreSetupView extends Component {
 	}
 }
 
-function mapStateToProps( /* state */ ) {
+function mapStateToProps( state ) {
+	const site = getSelectedSiteWithFallback( state );
+	let name = '';
+	if ( site ) {
+		name = getSiteTitle( state, site.ID );
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/dashboard/pre-setup-view.js]: You can reduce these 4 lines down to just one: `const name = site && getSiteTitle( state, site.ID ) || ''`.

File: client/extensions/woocommerce/app/dashboard/pre-setup-view.js [hunk 3]
```diff
@@ -10,46 +10,100 @@ import React, { Component, PropTypes } from 'react';
...
+		};
+
+		const onFailure = () => {
+			this.setState( { isSaving: false } );
+			return errorNotice( translate( 'There was a problem saving the store address. Please try again.' ) );
+		};
+
+		this.props.updateStoreAddress(
+			site.ID,
+			this.state.address.street,
```
Comment (belcherj) [inline client/extensions/woocommerce/app/dashboard/pre-setup-view.js]: doesn't spread operator help you here:  ...this.state.address ?

Comment (belcherj) [inline client/extensions/woocommerce/app/dashboard/pre-setup-view.js]: ignore, I guess you can assume it is ordered correctly.

Comment (allendav) [inline client/extensions/woocommerce/app/dashboard/pre-setup-view.js]: I was a little nervous relying on spread operator here - to keep the ordering in the call should the object construction change

**Comment by allendav** [comment]
ty all! will roll in a few feedbacks and then commit

File: client/extensions/woocommerce/app/dashboard/pre-setup-view.js (hunk 2 above)
Comment (allendav) [inline client/extensions/woocommerce/app/dashboard/pre-setup-view.js]: I'd like to keep it this way until I get more comfortable with using the && and || operator this way. I've seen some behavior lately that I don't completely understand.

File: client/extensions/woocommerce/app/dashboard/pre-setup-view.js (hunk 1 above)
Comment (allendav) [inline client/extensions/woocommerce/app/dashboard/pre-setup-view.js]: Thank you :) Fixed in https://github.com/Automattic/wp-calypso/pull/15463/commits/bc466ca37e62386bab22f2d209ed4199fce059dd

**Pull request merged**